### PR TITLE
fix(ButtonGroup): added sameHeight feature - Leadspace issue related

### PR DESCRIPTION
**Pull request by caiobozato**
### Related Ticket(s)

Pattern: Leadspace - Large text on button wraps to 2, 3 lines and makes only one of the buttons taller. #1574

### Description

The Buttons should have the same height, even if a text of one button is larger than the other. The issue mentioned above is related to `Leadspace` pattern, but the fix was applied to `ButtonGroup` since it's used in the Leadspace pattern.
 
### Changelog

**New**

- Added `sameHeight` utility to ButtonGroup

### CODE REVIEW

File: packages/react/src/components/ButtonGroup/ButtonGroup.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ import React, { useEffect, useLayoutEffect, useRef } from 'react';
 import Button from '../../internal/vendor/carbon-components-react/components/Button/Button';
 import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
 import PropTypes from 'prop-types';
+import root from 'window-or-global';
+import { sameHeight } from '@carbon/ibmdotcom-utilities';
```
Comment (annawen1) [inline packages/react/src/components/ButtonGroup/ButtonGroup.js:12]: I believe we are now importing exports like so

`import  sameHeight  from '@carbon/ibmdotcom-utilities/es/utilities/sameHeight/sameHeight';`

to optimize bundle size

File: packages/react/src/components/ButtonGroup/ButtonGroup.js [hunk 2]
```diff
@@ -97,6 +99,26 @@ const ButtonGroup = ({ buttons, enableSizeByContent }) => {
...
+  }, []);
+
+  /**
+   * Set the buttons to have the same height based on the tallest one
+   */
+  const setSameHeight = () => {
+    root.requestAnimationFrame(() => {
+      const { current: containerNode } = groupRef;
+      if (containerNode) {
+        sameHeight(
```
Comment (jeffchew) [inline packages/react/src/components/ButtonGroup/ButtonGroup.js:115]: @caiobozato @annawen1 wonder if it's worth adding logic to only do this when there are 2 or more buttons?

Comment (annawen1) [reply packages/react/src/components/ButtonGroup/ButtonGroup.js:115]: hmm good point we probably want to skip over all that logic if we're only using one button

**Comment by asudoh** [comment]
Nice try @caiobozato! Similar to #2714, wondering if we need to support dynamic change of the content (e.g. application sets a different button content in certain context or at some point):

leadspace-resize

File: packages/react/src/components/ButtonGroup/ButtonGroup.js (hunk 2 above)
Comment (caiobozato) [reply packages/react/src/components/ButtonGroup/ButtonGroup.js:117]: @annawen1 @jeffchew Conditional logic added in latest commit

**Comment by caiobozato** [comment]
> Nice try @caiobozato! Similar to #2714, wondering if we need to support dynamic change of the content (e.g. application sets a different button content in certain context or at some point):
> 
> leadspace-resize

@asudoh Like we talked on Slack, I saw your comments about `ResizeObserver` in #2714 , I think we can bring up this for discussion with DDS engineers team and see if wouldn’t be better to update the sameHeight utility to use the `ResizeObserver`

**Comment by jeffchew** [comment]
> @asudoh As we talked on Slack, I saw your comments about `ResizeObserver` in #2714 , I think we can bring up this for discussion with DDS engineers team and see if wouldn’t be better to update the sameHeight utility to use the `ResizeObserver`, this way we can standardize the API use.

Agreed, we should centralize the change in the utility so all components using it will take advantage of this.

**Pull request merged**